src/mon/OSDMonitor.cc

```diff
@@ -2265,6 +2258,24 @@ bool OSDMonitor::prepare_boot(MonOpRequestRef op)
     else
       xi.features = m->get_connection()->get_features();
 
+    // mark in?
+    if ((g_conf->mon_osd_auto_mark_auto_out_in &&
```

Comment:
The bug was that the original fix set new_xinfo without populating the entry with the previous map's xinfo structure, so ti clobbered other fields (in this case, features). In the fixed version we move the whole if down below where we have the xi already being properly populated and filled in with other info.